tools/serve/serve.py

```diff
@@ -345,6 +345,55 @@ class ServiceWorkerModulesHandler(HtmlWrapperHandler):
 </script>
 """
 
+class ShadowRealmHandler(HtmlWrapperHandler):
+    global_type = "shadowrealm"
+    path_replace = [(".any.shadowrealm.html", ".any.js")]
+
+    wrapper = """<!doctype html>
+<meta charset=utf-8>
+%(meta)s
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+(async function() {
+  const r = new ShadowRealm();
+  const results = JSON.parse(await new Promise(r.evaluate(`
+    (resolve, reject) => {
+      (async () => {
+        await import("/resources/testharness.js");
+        %(script)s
+        globalThis.self = globalThis;
```

Comment:
This should not be necessary, since in browsers I believe the plan is for `self` to continue working and return the global this value in shadow realms.

Reply:
Removed. I was trying to stick closely to the spec drafts that I'm aware of--do you have more details or a link to the discussion? I know the HTML side of the specification isn't finalized yet...

Reply:
Yeah, I guess that part isn't specified yet... only ErrorEvent and PromiseRejectionEvent are. Hmm. I think it's still best to write the test harness under the assumption that it will get specified this way though.